### ISSUE TITLE
Redshift: "EXCEPT" should be a synonym of "MINUS"

### ISSUE THREAD

**Issue by pyrr** [sqlfluff/sqlfluff#7169]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.


### What Happened

SQLFluff does not parse `EXCEPT` correctly in the Redshift dialect. However, SQLFluff handles `MINUS` correctly.

According to the [docs](https://docs.aws.amazon.com/redshift/latest/dg/r_UNION.html#r_UNION-parameters), "MINUS and EXCEPT are exact synonyms," so their behavior should be identical in SQLFluff.

### Expected Behaviour

### test_minus.sql
```sql
SELECT col FROM tab1
MINUS
(SELECT col FROM tab2)
;

```

### Lint (MINUS)
```
$ sqlfluff lint test_minus.sql 
All Finished 📜 🎉!
```

### Parse (MINUS)
```
$ sqlfluff parse test_minus.sql 
[L:  1, P:  1]      |file:
[L:  1, P:  1]      |    statement:
[L:  1, P:  1]      |        set_expression:
[L:  1, P:  1]      |            select_statement:
[L:  1, P:  1]      |                select_clause:
[L:  1, P:  1]      |                    keyword:                                  'SELECT'
[L:  1, P:  7]      |                    [META] indent:
[L:  1, P:  7]      |                    whitespace:                               ' '
[L:  1, P:  8]      |                    select_clause_element:
[L:  1, P:  8]      |                        column_reference:
[L:  1, P:  8]      |                            naked_identifier:                 'col'
[L:  1, P: 11]      |                    [META] dedent:
[L:  1, P: 11]      |                whitespace:                                   ' '
[L:  1, P: 12]      |                from_clause:
[L:  1, P: 12]      |                    keyword:                                  'FROM'
[L:  1, P: 16]      |                    whitespace:                               ' '
[L:  1, P: 17]      |                    from_expression:
[L:  1, P: 17]      |                        [META] indent:
[L:  1, P: 17]      |                        from_expression_element:
[L:  1, P: 17]      |                            table_expression:
[L:  1, P: 17]      |                                table_reference:
[L:  1, P: 17]      |                                    naked_identifier:         'tab1'
[L:  1, P: 21]      |                        [META] dedent:
[L:  1, P: 21]      |            newline:                                          '\n'
[L:  2, P:  1]      |            set_operator:
[L:  2, P:  1]      |                keyword:                                      'MINUS'
[L:  2, P:  6]      |            newline:                                          '\n'
[L:  3, P:  1]      |            bracketed:
[L:  3, P:  1]      |                start_bracket:                                '('
[L:  3, P:  2]      |                [META] indent:
[L:  3, P:  2]      |                select_statement:
[L:  3, P:  2]      |                    select_clause:
[L:  3, P:  2]      |                        keyword:                              'SELECT'
[L:  3, P:  8]      |                        [META] indent:
[L:  3, P:  8]      |                        whitespace:                           ' '
[L:  3, P:  9]      |                        select_clause_element:
[L:  3, P:  9]      |                            column_reference:
[L:  3, P:  9]      |                                naked_identifier:             'col'
[L:  3, P: 12]      |                        [META] dedent:
[L:  3, P: 12]      |                    whitespace:                               ' '
[L:  3, P: 13]      |                    from_clause:
[L:  3, P: 13]      |                        keyword:                              'FROM'
[L:  3, P: 17]      |                        whitespace:                           ' '
[L:  3, P: 18]      |                        from_expression:
[L:  3, P: 18]      |                            [META] indent:
[L:  3, P: 18]      |                            from_expression_element:
[L:  3, P: 18]      |                                table_expression:
[L:  3, P: 18]      |                                    table_reference:
[L:  3, P: 18]      |                                        naked_identifier:     'tab2'
[L:  3, P: 22]      |                            [META] dedent:
[L:  3, P: 22]      |                [META] dedent:
[L:  3, P: 22]      |                end_bracket:                                  ')'
[L:  3, P: 23]      |    newline:                                                  '\n'
[L:  4, P:  1]      |    statement_terminator:                                     ';'
[L:  4, P:  2]      |    newline:                                                  '\n'
[L:  5, P:  1]      |    [META] end_of_file:

```


### Observed Behaviour

### test_except.sql
```sql
SELECT col FROM tab1
EXCEPT
(SELECT col FROM tab2)
;

```

### Lint (EXCEPT)
```
$ sqlfluff lint test_except.sql 
== [test_except.sql] FAIL                                                                                                                                                                                                                                      
L:   2 | P:   1 |  PRS | Line 2, Position 1: Found unparsable section:
                       | 'EXCEPT\n(SELECT col FROM tab2)'
WARNING: Parsing errors found and dialect is set to 'redshift'. Have you configured your dialect correctly?
All Finished 📜 🎉!
```

### Parse (EXCEPT)
```
$ sqlfluff parse test_except.sql 
[L:  1, P:  1]      |file:
[L:  1, P:  1]      |    statement:
[L:  1, P:  1]      |        select_statement:
[L:  1, P:  1]      |            select_clause:
[L:  1, P:  1]      |                keyword:                                      'SELECT'
[L:  1, P:  7]      |                [META] indent:
[L:  1, P:  7]      |                whitespace:                                   ' '
[L:  1, P:  8]      |                select_clause_element:
[L:  1, P:  8]      |                    column_reference:
[L:  1, P:  8]      |                        naked_identifier:                     'col'
[L:  1, P: 11]      |                [META] dedent:
[L:  1, P: 11]      |            whitespace:                                       ' '
[L:  1, P: 12]      |            from_clause:
[L:  1, P: 12]      |                keyword:                                      'FROM'
[L:  1, P: 16]      |                whitespace:                                   ' '
[L:  1, P: 17]      |                from_expression:
[L:  1, P: 17]      |                    [META] indent:
[L:  1, P: 17]      |                    from_expression_element:
[L:  1, P: 17]      |                        table_expression:
[L:  1, P: 17]      |                            table_reference:
[L:  1, P: 17]      |                                naked_identifier:             'tab1'
[L:  1, P: 21]      |                    [META] dedent:
[L:  1, P: 21]      |            newline:                                          '\n'
[L:  2, P:  1]      |            unparsable:                                       !! Expected: 'Nothing here.'
[L:  2, P:  1]      |                word:                                         'EXCEPT'
[L:  2, P:  7]      |                newline:                                      '\n'
[L:  3, P:  1]      |                start_bracket:                                '('
[L:  3, P:  2]      |                word:                                         'SELECT'
[L:  3, P:  8]      |                whitespace:                                   ' '
[L:  3, P:  9]      |                word:                                         'col'
[L:  3, P: 12]      |                whitespace:                                   ' '
[L:  3, P: 13]      |                word:                                         'FROM'
[L:  3, P: 17]      |                whitespace:                                   ' '
[L:  3, P: 18]      |                word:                                         'tab2'
[L:  3, P: 22]      |                end_bracket:                                  ')'
[L:  3, P: 23]      |    newline:                                                  '\n'
[L:  4, P:  1]      |    statement_terminator:                                     ';'
[L:  4, P:  2]      |    newline:                                                  '\n'
[L:  5, P:  1]      |    [META] end_of_file:

==== parsing violations ====
L:   2 | P:   1 |  PRS | Line 2, Position 1: Found unparsable section: 'EXCEPT\n(SELECT col
                       | FROM tab2)'
WARNING: Parsing errors found and dialect is set to 'redshift'. Have you configured your dialect correctly?
```

### How to reproduce

Write any statement containing `EXCEPT` with the Redshift dialect.

### Dialect

Redshift

### Version

```
$ sqlfluff --version
sqlfluff, version 3.4.2

$ python -V
Python 3.11.0
```

### Configuration

```
[sqlfluff]
dialect = redshift
max_line_length = 180
large_file_skip_byte_limit = 0
processes = 0
exclude_rules = LT08,RF02,AL03,AM04,ST07,ST06,ST09
warnings = RF04,AM01,AM02,AM03,AM05,AM06,AM07,AM08,RF05

[sqlfluff:rules:capitalisation.keywords]
capitalisation_policy = upper

[sqlfluff:rules:capitalisation.identifiers]
extended_capitalisation_policy = lower

[sqlfluff:rules:capitalisation.functions]
extended_capitalisation_policy = upper

[sqlfluff:rules:capitalisation.literals]
capitalisation_policy = upper

[sqlfluff:rules:capitalisation.types]
extended_capitalisation_policy = upper

[sqlfluff:rules:convention.terminator]
multiline_newline = True
require_final_semicolon = True

[sqlfluff:rules:layout.newlines]
maximum_empty_lines_between_statements = 0
maximum_empty_lines_inside_statements = 0

[sqlfluff:rules:layout.long_lines]
ignore_comment_lines = True
```

### Are you willing to work on and submit a PR to address the issue?

- [x] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by franloza** [comment]
Hi! 👋 I opened a PR to fix this issue: https://github.com/sqlfluff/sqlfluff/pull/7184
